app/assets/stylesheets/zammad.scss

```diff
@@ -6894,6 +6894,20 @@ a.list-group-item.active > .badge,
   display: none;
 }
 
+.article-add {
+  .article-visibility-text {
+    display: none;
+  }
+
+  &.is-open.is-internal .article-visibility-text[data-type='internal-note'] {
+    display: block;
+  }
+
+  &.is-open.is-public .article-visibility-text[data-type='public-note'] {
+    display: block;
+  }
+}
+
```

Comment:
@fflorent this looks like a very fragile way to handle this.

I'd rather extend [App.TicketZoomArticleNew#setArticleInternal](https://github.com/zammad/zammad/blob/5c1cbb966ea6e4a19414d6b6cfb95f7fa0add254/app/assets/javascripts/app/controllers/ticket_zoom/article_new.coffee#L367) along the spirit of [App.TicketZoomArticleNew#setArticleTypePre](https://github.com/zammad/zammad/blob/5c1cbb966ea6e4a19414d6b6cfb95f7fa0add254/app/assets/javascripts/app/controllers/ticket_zoom/article_new.coffee#L376)

Reply:
Why would that be fragile?

Reply:
Please take this with a grain of salt since it's project-specific :) It may be totally fine (and even preferred) in another codebase where it's CSS-first.

We usually do this sort of switching in Coffeescript event callbacks. Finding elements based on `data-*` or classes and then manipulating them. This allows to put all UI manipulations to related callbacks.

When a part of this goes directly into CSS, somebody editing this area must remember to also look into CSS. Aside from just checking out related callbacks.

If this was relying on new article data type and visibility, we'd have seen early on that there're 3 types all of which can be public or private. Meanwhile this approach allows to easily use `-note` in context of phone/email. Which is a bit ambiguous. `note` is only 1 of 3 available `article` types.

On top of that, this creates a lot of repetitive CSS. We'd have to have 6 nearly identical CSS selectors to cover all types/visibility pairs.

Reply:
Happy new year!

Thanks :). I didn't take personally your remark (it could have been truly a bad piece of code), but I could not understand what were the reasons that led you state the code was fragile.

Your point is fair, I take into account this remark

Reply:
I think it's done